src/compiler/factory.ts

```diff
@@ -1618,17 +1618,23 @@ namespace ts {
         );
     }
 
-    function createReactNamespace(reactNamespace: string, parent: JsxOpeningLikeElement) {
+    export function createJsxFactory(jsxFactory: string) {
+        // No explicit validation of this parameter is required. Users are

```

Comment:
do not think you can do this. in Program:verifyCompilerOptions, you should check that `isIdentifierText(options.jsxFactory, languageVersion)`